lib/vfscore/pipe.c

```diff
@@ -604,10 +604,12 @@ UK_SYSCALL_R_DEFINE(int, pipe2, int*, pipefd, int, flags)
 	return 0;
 }
 
+#if UK_LIBC_SYSCALLS
 /* TODO maybe find a better place for this when it will be implemented */
 int mkfifo(const char *path __unused, mode_t mode __unused)
```

Comment:
Wouldn't make using `UK_SYSCALL_R_DEFINE` instead, more sense?

Reply:
`mkfifo` is not a syscall. Musl will do:
```
int mkfifo(const char *path, mode_t mode)
{
        return mknod(path, mode | S_IFIFO, 0);
}
```
And newlib will use the definition above.

Reply:
Fair enough :+1: